pkg/reconciler/v1alpha1/taskrun/resources/taskresourceresolution_test.go

```diff
@@ -192,3 +189,42 @@ func TestResolveTaskRun_noResources(t *testing.T) {
 		t.Errorf("Did not expect any outputs to be resolved when none specified but had %v", rtr.Outputs)
 	}
 }
+
+type testGetter struct {
+	err bool
+}
+
+func (t testGetter) Get(n string) (*v1alpha1.PipelineResource, error) {
+	if t.err {
+		return nil, fmt.Errorf("nope")
+	} else {
+		return &v1alpha1.PipelineResource{}, nil
+	}
+}
+
+var resourceIndex int
+
+type resourceGetter struct{}
+
+func (r resourceGetter) Get(n string) (*v1alpha1.PipelineResource, error) {
```

Comment:
if we're going to start making these Getters more complex, maybe we can put them in a lib in `test/`?

Reply:
I thought they could be defined where they are being used. Are you talking about some kind of test helpers?

Reply:
> I thought they could be defined where they are being used. Are you talking about some kind of test helpers?

yeah exactly but you know what, i take it back, i think what you've done makes more sense :D just ignore me (if we start making these helpers more complicated they're going to be hard to deal with... and we'll need tests for them... X'D)